sda/cmd/s3inbox/proxy.go

```diff
@@ -199,6 +200,27 @@ func (p *Proxy) allowedResponse(w http.ResponseWriter, r *http.Request, token jw
 			return
 		}
 
+		// The following block is for treating the case when the client loses connection to the server and then it reconnects to a
+		// different instance of s3inbox. For more details see https://github.com/NBISweden/BigPicture-Deployment/issues/283.
+		if p.fileIds[r.URL.Path] == "" {
+
+			corrID, err := p.database.GetCorrID(username, filepath, "")
+			if err != nil {
+				p.internalServerError(w, r, fmt.Sprintf("failed to retrieve corrID from database: %v", err))
+
+				return
+			}
+
+			p.fileIds[r.URL.Path], err = p.database.GetFileID(corrID)
+			if err != nil {
+				p.internalServerError(w, r, fmt.Sprintf("failed to retrieve fileID from database: %v", err))
+
+				return
+			}
+
+			log.Debugf("resuming work on file with fileId: %v", p.fileIds[r.URL.Path])
+		}
+
```

Comment:
```suggestion
		corrID, err := p.database.GetCorrID(username, filepath, "")
		if err != nil {
			p.internalServerError(w, r, fmt.Sprintf("failed to retrieve corrID from database: %v", err))

			return
		}

		p.fileIds[r.URL.Path], err = p.database.GetFileID(corrID)
		if err != nil {
			p.internalServerError(w, r, fmt.Sprintf("failed to retrieve fileID from database: %v", err))

			return
		}
		log.Debugf("resuming work on file with fileId: %v", p.fileIds[r.URL.Path])
```

Technically we only need this since `fileIds` is a `map[string]sting` and as such can not hold duplicate entries.

One thing we need to take into account is memory leak, unless the `p.fileIds` is cleaned it will continue to increase in size until the app is restarted.

Reply:
The reason why I added the if clause is to avoid a couple of extra calls to the database when they are not needed. So I would suggest we keep a bit of more complexity in favor of performance, but otherwise I agree that it is not strictly needed.

Good point about the memory leak. Worst case, won't the pod become OOM and restarted by k8s?

Reply:
>Good point about the memory leak. Worst case, won't the pod become OOM and restarted by k8s?

Yes, which might break the upload for someone.

Reply:
>The reason why I added the if clause is to avoid a couple of extra calls to the database when they are not needed. So I would suggest we keep a bit of more complexity in favor of performance, but otherwise I agree that it is not strictly needed.

These two calls only happen when a file uploaded is completed, it's not going to decrease the performance.

Reply:
> > Good point about the memory leak. Worst case, won't the pod become OOM and restarted by k8s?
> 
> Yes, which might break the upload for someone.

indeed, but with this fix the upload should continue with the other pod (if we have deployed one). We should fixing eitherway, I am just theoritizing here :-)

Reply:
> > The reason why I added the if clause is to avoid a couple of extra calls to the database when they are not needed. So I would suggest we keep a bit of more complexity in favor of performance, but otherwise I agree that it is not strictly needed.
> 
> These two calls only happen when a file uploaded is completed, it's not going to decrease the performance.

I still don't see why add two more calls to the database when they are not needed instead of a cheap check for an empty field.

Reply:
> One thing we need to take into account is memory leak, unless the `p.fileIds` is cleaned it will continue to increase in size until the app is restarted. Which will happen if the start and end action is on different pods.

I just noticed that a cleanup is done a bit below, [here](https://github.com/neicnordic/sensitive-data-archive/blob/591e5d0c7656410cf617a94589cf0919b959eb15/sda/cmd/s3inbox/proxy.go#L210). This looks enough?

Reply:
That is still inside the `completed uploads` block